packages/atomic/src/components/search/atomic-result-template/atomic-result-template.spec.ts

```diff
@@ -0,0 +1,116 @@
+import type {Result} from '@coveo/headless';
+import {ResultTemplatesHelpers} from '@coveo/headless';
+import {html} from 'lit';
+import {describe, expect, it, vi} from 'vitest';
+import {makeMatchConditions} from '@/src/components/common/template-controller/template-utils';
+import {fixture} from '@/vitest-utils/testing-helpers/fixture';
+import {sanitizeHtml} from '@/vitest-utils/testing-helpers/testing-utils/sanitize-html';
+import {AtomicResultTemplate} from './atomic-result-template.js';
+
+vi.mock('@/src/components/common/template-controller/template-utils', {
+  spy: true,
+});
+
+describe('atomic-result-template', () => {
+  type AtomicResultTemplateProps = Pick<
+    AtomicResultTemplate,
+    'conditions' | 'mustMatch' | 'mustNotMatch'
+  >;
+
+  const setupElement = async (
+    options: Partial<AtomicResultTemplateProps> = {}
+  ) => {
+    const defaultProps: AtomicResultTemplateProps = {
+      conditions: [],
+      mustMatch: {},
+      mustNotMatch: {},
+    };
+
+    const container = document.createElement('atomic-result-list');
+    const element = await fixture<AtomicResultTemplate>(
+      html`
+        <atomic-result-template
+          .conditions=${options.conditions || defaultProps.conditions}
+          .mustMatch=${options.mustMatch || defaultProps.mustMatch}
+          .mustNotMatch=${options.mustNotMatch || defaultProps.mustNotMatch}
+        >
+          <template slot="default">
+            <div>Result Template Content</div>
+          </template>
+        </atomic-result-template>
+      `,
+      container
+    );
+    return element;
+  };
+
+  it('should instantiate without errors', async () => {
+    const element = await setupElement();
+    expect(element).toBeInstanceOf(AtomicResultTemplate);
+  });
+
+  it('should have default empty mustMatch, mustNotMatch and conditions', async () => {
+    const element = await setupElement();
+    expect(element.mustMatch).toEqual({});
+    expect(element.mustNotMatch).toEqual({});
+    expect(element.conditions).toEqual([]);
+  });
+
+  describe('when must-match and must-not-match attributes are set', () => {
+    it('should call #makeMatchConditions on connectedCallback', async () => {
```

Comment:
Also test that ResultTemplateController.prototype, 'matchConditions', 'set' is called with the correct arg.